pandas/core/internals/blocks.py

```diff
@@ -180,7 +184,19 @@ def _check_ndim(self, values, ndim):
         ValueError : the number of dimensions do not match
         """
         if ndim is None:
-            ndim = values.ndim
+            warnings.warn(
+                "Accepting ndim=None in the Block constructor is deprecated, "
+                "this will raise in a future version.",
+                FutureWarning,
```

Comment:
normally i want FutureWarning, but in this case this is only an internal library use. ok to use DeprecationWarning here